File: java_algorithms/InsertionSort.java
package algorithms.java_algorithms;

public class InsertionSort<T extends Comparable<T>> {
    
    public void sort(T[] item) {
        
        for (int i = 1; i < item.length; i++) {

            T auxElement = item[i];
            int k = i;

            while (k > 0 && item[k - 1].compareTo(auxElement) > 0) {

                item[k] = item[k-1];
                k--;
            }

            item[k] = auxElement;
        }
    }
}

File: python/insertion_sort.py
def insertion_sort(item):
    
    for i in range(1, len(item)):

        auxElement = item[i]

        k = i

        while (k > 0 and item[k-1] > auxElement):

            item[k] = item[k-1]

            k -= 1

        item[k] = auxElement
